tests_new/integration_tests/modules/dashboards/test_dashboard.py

```diff
@@ -0,0 +1,113 @@
+from assertpy import assert_that
+
+from integration_tests.modules.dashboards.queries import (
+    search_dashboards,
+    get_dashboard,
+    list_dashboard_shares,
+    get_author_session,
+    get_reader_session,
+)
+from integration_tests.modules.dashboards.mutations import (
+    update_dashboard,
+    delete_dashboard,
+    approve_dashboard_share,
+    reject_dashboard_share,
+)
+from integration_tests.modules.dashboards.conftest import create_dataall_dashboard
+from integration_tests.errors import GqlError
+
+UPDATED_DESC = 'new description'
+
+
```

Comment:
I see that we are using the persistent environment to test the dashboards, same way for ml_studio (FYI @petrkalos); but for s3_datasets and notebooks we use session environments. I think we should agree on which one to use in the modules.

The main difference lies in the cdk extensions deployed as part of the environment stack: inside some modules (ml_studio, s3_datasets) we deployed additional infra as part of the environment stack (mlstudio domain; s3/glue custom resources). If we use the session environment we use the latest version of these "cdk-extensions" and we test how these are deployed from scratch. If we use the persistent environment we need to update the environment first and check it suceeded (which tests the backwards compatibility of the "cdk-extension") and then we can perform the test

In the most perfect scenario we would test the fresh-functionality of the module on one side and the backwards compatibility on the other. But it might be a bit too much. wdyt?

Reply:
In the most perfect I would say `persistent_env` should have an existing data.all resource of each type to update/test on and then `session_env` stands up a new one, tests, and tears down as well (and in a perfect world it all runs super fast 😄) 

For us I think the high priority items may need that type of testing (both persistent and session), including:
- S3 and Redshift Datasets
- Shares

For the play tool modules I think it is good to newly create, test, and tear down each time. Really this could happen in either session or persistent and has no real difference unless the module impacts Env CDK (which is ML Studio and Datasets only curently). Our currently set up is:
- ML Studio (currently does persistent **)
- Notebooks (currently does both persistent and session - might as well keep)
- Dashboards (currently does persistent)
- Data Pipelines (not implemented yet)
- Worksheets (N/A - not env or IaC related to creation)
- Omics  (not implemented yet)


** To note - for ML Studio we do stand up new user profiles, test and then tear down but do so in `persistent_env1`. I would think largely because the long run time needed to stand up new ML Studio Domain as IaC in env CDK and the fact that deleting a studio Domain right now does not fully clean up after itself (EFS left behind preventing VPC deletion)

Reply:
My suggestion is that we use session at a minimum everywhere (for play tools) and do a full create, test, and clean up -- except for MLStudio until we do some looking into some of the above

And for the high priority should have both session and persistent

I can switch dashboards to session_env and re-test (it should not change anything and would validate some of my pre-reqs again) 

Thoughts? @dlpzx

Reply:
I like the conclusions :) 
- splitting for Datasets/Shares(high-prio, infra dependent) vs Play tools
- Use session at a minimum
- Look at special cases with care - ML Studio dirty teardown of resources